synapse/config/oidc_config.py

```diff
@@ -191,6 +192,27 @@ def generate_config_section(self, config_dir_path, server_name, **kwargs):
         #           which is set to the claims returned by the UserInfo Endpoint and/or
         #           in the ID Token.
         #
+        #   It is possible to configure Synapse to only allow logins if certain attributes
+        #   match particular values in the OIDC userinfo. The requirements can be listed under
+        #   `attribute_requirements` as shown below. All of the listed attributes must
+        #   match for the login to be permitted. Additional attributes can be added to
+        #   userinfo by expanding the `scopes` section of the OIDC config to retrieve
+        #   additional information from the OIDC provider.
+        #
+        #   An important thing to note here, is that OIDC claims which exist in the userinfo
+        #   as a string will be strictly matched - the attribute matches only if the userinfo
+        #   value exactly matches the configured value in `attribute_requirements`.
+        #   However, if the claim exists in the userinfo as a list, the attribute matches
+        #   if the list contains the value configured in `attribute_requirements`.
+        #   i.e. `family_name` in the userinfo MUST be "Stephensson" in the example below
+        #   i.e. `groups` in the userinfo MUST contain "admin" in the example below
```

Comment:
Does this seem a bit simpler, but still accurate?

```suggestion
        #   If the OIDC claim is a list, then the attribute must match any value in the list.
        #   Otherwise, it must exactly match the value of the claim. Using the example
        #   below, the `family_name` claim MUST be "Stephensson", but the `groups`
        #   claim MUST contain "admin".
```

Reply:
Yeah, that's less wordy but still entirely accurate. Seems good!

Reply:
Woot! Thanks for making this change!